mongo/options/clientoptions_test.go

```diff
@@ -573,6 +573,14 @@ func TestClientOptions(t *testing.T) {
 				"mongodb://localhost/?tlsCertificateKeyFile=testdata/one-pk-multiple-certs.pem",
 				baseClient().SetTLSConfig(&tls.Config{Certificates: make([]tls.Certificate, 1)}),
 			},
+			{
+				"GODRIVER-2650 X509 certificate",
+				"mongodb://localhost/?ssl=true&authMechanism=mongodb-x509&sslClientCertificateKeyFile=testdata/one-pk-multiple-certs.pem",
+				baseClient().SetAuth(Credential{
+					AuthMechanism: "mongodb-x509", AuthSource: "$external",
```

Comment:
Just to verify my understanding: I assume this test is trying to certify that we are taking [the first certificate](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/options/testdata/one-pk-multiple-certs.pem#L36-L56) in `one-pk-multiple-certs` instead of the second. Running the two certs in that file through a certificate decoder, I see the only difference between them is the common name (`CN`), which is "localhost" and "client-intermediate" for the first and second respectively. Are we relying on the `Username` field on `Credential` to not match in its `CN` if we take the second cert instead of the first?

Reply:
Yes for this PEM file. We rely only on the `CN` section to verify that we pick up the correct subject name.